projects/igniteui-angular/src/lib/directives/for-of/for_of.directive.ts

```diff
@@ -1472,9 +1460,14 @@ export class IgxForOfDirective<T, U extends T[] = T[]> extends IgxForOfToken<T,U
 
     private _updateHScrollOffset() {
         let scrollOffset = 0;
-        scrollOffset = this.scrollComponent.nativeElement &&
-            this.scrollComponent.size ?
-            this.scrollPosition - this.sizesCache[this.state.startIndex] : 0;
+        let currentScrollTop = this.scrollPosition;
```

Comment:
This and the previous method can be unified and only based on direction set top or left.